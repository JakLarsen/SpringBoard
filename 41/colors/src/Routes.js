import {Route, Redirect} from 'react-router-dom'

import Color from './Color'
import ColorForm from './ColorForm'
import Home from './Home'




const Routes = () => {

    return (
        <>
            {/* Home isn't rendering?? when we have the Redirect?? */}
            <Route exact path='/'>
                <Home/>
            </Route>
            <Route exact path='/colors'>
                <ColorForm/>
            </Route>
            <Route exact path='/colors/:color'>
                <ColorForm/>
                <Color/>
            </Route>
            <Redirect to='/colors'/>
        </>
    )
}

export default Routes