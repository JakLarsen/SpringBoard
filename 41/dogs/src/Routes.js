import {BrowserRouter, Route, Switch, Redirect} from 'react-router-dom'

import Dogs from './Dogs';
import Dog from './Dog';



const Routes = () => {
     return (
         <>
            <Route exact path='/dogs'><Dogs/></Route>
            <Route exact path='/dogs/:name'><Dog/></Route>
            <Redirect to='/dogs'/>
        </>
     )
}

export default Routes