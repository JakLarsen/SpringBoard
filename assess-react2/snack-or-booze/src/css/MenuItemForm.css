.menu-item-form{
    position: relative;
    display: flex;
    flex-flow: column nowrap;
}