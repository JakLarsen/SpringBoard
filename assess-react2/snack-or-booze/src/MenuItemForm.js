import React, {useState} from 'react'
import {
    Card,
    CardBody,
    CardTitle,
    CardText,
  } from "reactstrap";
import './css/MenuItemForm.css'



/**
 * Form to add a snack or drink to the menu
 * (Only using State currently, not our pseudo-database)
 */

const MenuItemForm = ({title, addSnack, addDrink}) => {

    const INITIAL_STATE = {
        id: "",
        itemType: "",
        name: "",
        description: "",
        recipe: "",
        serve: ""
    }

    const [formData, setFormData] = useState(INITIAL_STATE)

    //A toggle to see if we're adding a snack or drink
    let addFunctionToUse = (
        title === 'Snacks' ? addSnack : addDrink
    )

    /**
     * onChange controlled-handler for all MenuItemForm input fields
     */
    const handleChange = (e) => {
        const {name, value} = e.target
        setFormData(formData => (
            {
                ...formData,
                [name]: value
            }
        ))
    }

    /**
     * onSubmit handler for our form
     * Uses addSnack or addDrink as addFunctionToUse based which type
     * Resets form values on submit
     */
    const handleSubmit = (e) => {
        e.preventDefault();
        console.log('SUBMITTED')
        //Could add to database as well, not just state
        //For some reason the fireEvent click doesn't think addFunctionToUse is a function
        addFunctionToUse({...formData})
        setFormData(INITIAL_STATE)
    }

    return (
        <div>
            <Card>
                <CardBody>
                    <CardTitle className="font-weight-bold text-center">
                        {`Add ${title} Form`}
                    </CardTitle>
                    <CardText>
                        Add a new item to the menu!
                    </CardText>

                    <form className="menu-item-form" onSubmit={handleSubmit}>
                        <label htmlFor="item-type">Category: </label>
                        <input 
                            id="itemType" name="itemType" 
                            type="text" placeholder="snacks or drinks" 
                            value={formData.itemType} onChange={handleChange}
                        />
                        <label htmlFor="name">Name:</label>
                        <input 
                            id='name' name='name' 
                            placeholder="item name"
                            value={formData.name} onChange={handleChange}  
                        />
                        <label htmlFor="id">ID:</label>
                        <input 
                            id='id' name='id' 
                            placeholder="item id"
                            value={formData.id} onChange={handleChange}  
                        />
                        <label htmlFor="description">Description:</label>
                        <input 
                            id='description' name='description' 
                            placeholder="A delicious snack"
                            value={formData.description} onChange={handleChange}  
                        />
                        <label htmlFor="recipe">Recipe:</label>
                        <input 
                            id='recipe' name='recipe' 
                            placeholder="cheese whizz and potatoes" 
                            value={formData.recipe} onChange={handleChange} 
                        />
                        <label htmlFor="serve">Serve:</label>
                        <input 
                            id='serve' name='serve' 
                            placeholder="ice cold"
                            value={formData.serve} onChange={handleChange} 
                        />
                        <button>Add item</button>
                    </form>   

                </CardBody>
            </Card>
        </div>
    )
}

export default MenuItemForm
