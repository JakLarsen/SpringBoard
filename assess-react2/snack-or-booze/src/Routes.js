import React, { useState, useEffect } from "react";
import "./css/Routes.css";
import Home from "./Home";
import SnackOrBoozeApi from "./Api";
import { Route, Switch} from "react-router-dom";
import Menu from "./Menu";
import MenuItem from "./MenuItem";
import NotFound from "./NotFound";
import Order from "./Order";
import OrderMenu from "./OrderMenu";

/**
 * Our main Routes, State Management, and Handlers
 */
function Routes() {

 /**
  * State Management
  * 
  * -Loading
  * -snacks
  * -drinks
  * -order
  */
  const [isLoading, setIsLoading] = useState(true);
  const [snacks, setSnacks] = useState([]);
  const [drinks, setDrinks] = useState([]);
  let [order, setOrder] = useState([])


  /**
   * useEffect for retrieving snacks, drinks, and order on initial load
   */
  useEffect(() => {
    async function getSnacks() {
      let snacks = await SnackOrBoozeApi.getSnacks();
      setSnacks(snacks);
      setIsLoading(false);
    }
    async function getDrinks() {
      let drinks = await SnackOrBoozeApi.getDrinks();
      setDrinks(drinks);
      setIsLoading(false);
    }
    async function getOrder(){
      let order = await SnackOrBoozeApi.getOrder();
      setOrder(order)
      setIsLoading(false)
    }
    getSnacks();
    getDrinks();
    getOrder();
  }, []);

  const addSnack = (snack) => {
      setSnacks(snacks => [...snacks, snack ])
      //Could post snack as well
  }
  const addDrink = (drink) => {
      setDrinks(drinks => [...drinks, drink ])
      //Could post drink as well
  }
  const addToOrder = (item) => {
    setOrder(items => [...items, item])
      //Order I thnk I'd keep solely as state while on the site for now
  }

  /**
   * Remove the first item with a given id from your order
   */
  const removeFromOrder = (id) => {

    let yetToRemove = true
    let newOrder = order.map(x=>x)

    newOrder.map((item, idx)=> {
      if(item.id === id && yetToRemove){
        newOrder.splice(idx, 1) 
        yetToRemove = false
      } 
    })
    setOrder(newOrder)
  }

  if (isLoading) {
    return <p>Loading &hellip;</p>;
  }

  return (
    <div className="Routes">
      <Switch>
        <Route exact path="/">
            <Home />
        </Route>
        <Route exact path="/snacks">
            <Menu items={snacks} title="Snacks" addSnack={addSnack}/>
        </Route>
        <Route path="/snacks/:id">
            <MenuItem items={snacks} cantFind="/snacks" />
        </Route>
        <Route exact path="/drinks">
            <Menu items={drinks} title="Drinks" addDrink={addDrink}/>
        </Route>
        <Route path="/drinks/:id">
            <MenuItem items={drinks} cantFind="/drinks" />
        </Route>
        <Route path="/order">
            <Order items={order} title="Order" removeFromOrder={removeFromOrder}/>
            <OrderMenu items={snacks} title="Snacks" addToOrder={addToOrder} />
            <OrderMenu items={drinks} title="Drinks" addToOrder={addToOrder} />
        </Route>
        <Route component={NotFound}/>
      </Switch>
    </div>
  );
}

export default Routes;
