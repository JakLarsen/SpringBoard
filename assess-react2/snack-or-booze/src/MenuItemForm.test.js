import React from 'react';
import { render, screen } from '@testing-library/react';
import MenuItemForm from './MenuItemForm'
import {addSnack, addDrink} from './Routes'

let title = "Snacks"

test("renders without crashing", ()=>{
    
    render(
        <MenuItemForm title={title} addSnack={addSnack} addDrink={addDrink}/>
    )
})

it('Matches snapshot', function(){
    const {asFragment} = render(
        <MenuItemForm title={title} addSnack={addSnack} addDrink={addDrink}/>
    )
    expect(asFragment()).toMatchSnapshot()
});