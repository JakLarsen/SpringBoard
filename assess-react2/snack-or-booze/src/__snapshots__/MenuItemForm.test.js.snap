// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <h5
          class="font-weight-bold text-center card-title"
        >
          Add Snacks Form
        </h5>
        <p
          class="card-text"
        >
          Add a new item to the menu!
        </p>
        <form
          class="menu-item-form"
        >
          <label
            for="item-type"
          >
            Category: 
          </label>
          <input
            id="itemType"
            name="itemType"
            placeholder="snacks or drinks"
            type="text"
            value=""
          />
          <label
            for="name"
          >
            Name:
          </label>
          <input
            id="name"
            name="name"
            placeholder="item name"
            value=""
          />
          <label
            for="id"
          >
            ID:
          </label>
          <input
            id="id"
            name="id"
            placeholder="item id"
            value=""
          />
          <label
            for="description"
          >
            Description:
          </label>
          <input
            id="description"
            name="description"
            placeholder="A delicious snack"
            value=""
          />
          <label
            for="recipe"
          >
            Recipe:
          </label>
          <input
            id="recipe"
            name="recipe"
            placeholder="cheese whizz and potatoes"
            value=""
          />
          <label
            for="serve"
          >
            Serve:
          </label>
          <input
            id="serve"
            name="serve"
            placeholder="ice cold"
            value=""
          />
          <button>
            Add item
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
